test/invalid-config.integration-test.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const shell = require('shelljs');
+const fs = require('fs');
+
+const mockers = require('./test-utils/mockers');
+
+function exec() {
+  const cli = require('../command');
+  const opt = cli.parse('commit-and-tag-version');
+  opt.skip = { commit: true, tag: true };
+  return require('../index')(opt);
+}
+
+/**
+ * Mock external conventional-changelog modules
+ *
+ * Mocks should be unregistered in test cleanup by calling unmock()
+ *
+ * bump?: 'major' | 'minor' | 'patch' | Error | (opt, parserOpts, cb) => { cb(err) | cb(null, { releaseType }) }
+ * changelog?: string | Error | Array<string | Error | (opt) => string | null>
+ * tags?: string[] | Error
+ */
+function mock({ bump, changelog, tags } = {}) {
+  mockers.mockRecommendedBump({ bump });
+
+  if (!Array.isArray(changelog)) changelog = [changelog];
+
+  mockers.mockConventionalChangelog({ changelog });
+
+  mockers.mockGitSemverTags({ tags });
+}
+
+// Really picky about running - works fine when run individually or in it's on file/jest-runner
+// but falls over if run as part of all tests in config-files.test
```

Comment:
If I were editing this test in the future, this comment makes me want to reach out to the person who wrote it to find out what they saw / suspected about ways to fix it, but that's usually impossible with an open source library

Could you add a bit about how it falls over (and optionally any suspicions about ways to sort it out, if you have them)?

Also - thanks for doing all this work, I'm sure this kind of thing made it very frustrating!

Reply:
Yup absolutely